reporter-sqs/src/main/java/zipkin/reporter/awssqs/AwsBufferedSqsSender.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright 2016 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.reporter.awssqs;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.handlers.AsyncHandler;
+import com.amazonaws.services.sqs.AmazonSQSAsyncClient;
+import com.amazonaws.services.sqs.buffered.AmazonSQSBufferedAsyncClient;
+import com.amazonaws.services.sqs.model.MessageAttributeValue;
+import com.amazonaws.services.sqs.model.SendMessageRequest;
+import com.amazonaws.services.sqs.model.SendMessageResult;
+import com.google.auto.value.AutoValue;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import zipkin.internal.LazyCloseable;
+import zipkin.internal.Nullable;
+import zipkin.reporter.BytesMessageEncoder;
+import zipkin.reporter.Callback;
+import zipkin.reporter.Encoding;
+import zipkin.reporter.Sender;
+
+import static zipkin.internal.Util.checkNotNull;
+
+/**
+ * This sends (usually TBinaryProtocol big-endian) encoded spans to an SQS queue
+ */
+@AutoValue
+public abstract class AwsBufferedSqsSender extends LazyCloseable<AmazonSQSBufferedAsyncClient> implements Sender {
+
+  private static final AtomicInteger messageId = new AtomicInteger(0);
+
+  public static AwsBufferedSqsSender create(String url) {
+    return builder().queueUrl(url).build();
+  }
+
+  public static Builder builder() {
+    return new AutoValue_AwsBufferedSqsSender.Builder()
+        .encoding(Encoding.THRIFT)
+        .compressionEnabled(true)
+        .credentialsProvider(new DefaultAWSCredentialsProviderChain())
+        .messageMaxBytes(256 * 1024 - 512); // 256KB SQS limit.  Leave some slack for attributes.
+  }
+
+  @AutoValue.Builder
+  public static abstract class Builder {
+
+    public abstract Builder queueUrl(String queueUrl);
+    public abstract Builder compressionEnabled(boolean compressionEnabled);
+    public abstract Builder dispatchExecutor(ExecutorService dispatchExecutor);
+    public abstract Builder credentialsProvider(AWSCredentialsProvider credentialsProvider);
+    public abstract Builder clientConfiguration(ClientConfiguration clientConfiguration);
```

Comment:
Ok I will remove the ClientConfiguration.  For most people the default AwsCredentialsProvider is sufficient though it is something we need to customized based on the way we handle IAM permissions.